assets/armsnippets.jsonc

```diff
@@ -142,6 +142,111 @@
+            "\t\"properties\": {",
+            "\t\t\"sku\": {",
+            "\t\t\t\"name\": \"${4|Standard_Small,Standard_Medium,Standard_Large,WAF_Medium,WAF_Large,Standard_v2,WAF_v2|}\",",
+            "\t\t\t\"tier\": \"${5|Standard,WAF,Standard_v2,WAF_v2|}\",",
+            "\t\t\t\"capacity\": 2",
+            "\t\t},",
+            "\t\t\"gatewayIPConfigurations\": [",
+            "\t\t\t{",
+            "\t\t\t\t\"name\": \"${6:appGatewayIpConfig}\",",
+            "\t\t\t\t\"properties\": {",
+            "\t\t\t\t\t\"subnet\": {",
+            "\t\t\t\t\t\t\t\"id\": \"[resourceId('Microsoft.Network/virtualNetworks/subnets', '${2:virtualNetwork1}', 'Subnet-1')]\"",
+            "\t\t\t\t\t}",
+            "\t\t\t\t}",
+            "\t\t\t}",
+            "\t\t],",
+            "\t\t\"frontendIPConfigurations\": [",
+            "\t\t\t{",
+            "\t\t\t\t\"name\": \"${7:appGatewayFrontendIP}\",",
+            "\t\t\t\t\"properties\": {",
+            "\t\t\t\t\t\"PublicIPAddress\": {",
+            "\t\t\t\t\t\t\"id\": \"[resourceId('Microsoft.Network/publicIPAddresses', '${3:publicIPAddress1}')]\"",
+            "\t\t\t\t\t}",
+            "\t\t\t\t}",
+            "\t\t\t}",
+            "\t\t],",
+            "\t\t\"frontendPorts\": [",
+            "\t\t\t{",
+            "\t\t\t\t\"name\": \"${8:appGatewayFrontendPort}\",",
+            "\t\t\t\t\"properties\": {",
+            "\t\t\t\t\t\"Port\": ${9:80}",
+            "\t\t\t\t}",
+            "\t\t\t}",
+            "\t\t],",
+            "\t\t\"backendAddressPools\": [",
+            "\t\t\t{",
+            "\t\t\t\t\"name\": \"${10:appGatewayBackendPool}\",",
+            "\t\t\t\t\"properties\": {}",
+            "\t\t\t}",
+            "\t\t],",
+            "\t\t\"backendHttpSettingsCollection\": [",
+            "\t\t\t{",
+            "\t\t\t\t\"name\": \"${13:appGatewayBackendHttpSettings}\",",
+            "\t\t\t\t\"properties\": {",
+            "\t\t\t\t\t\"Port\": ${14:80},",
+            "\t\t\t\t\t\"Protocol\": \"${15|Http,Https|}\",",
+            "\t\t\t\t\t\"CookieBasedAffinity\": \"Disabled\"",
+            "\t\t\t\t}",
+            "\t\t\t}",
+            "\t\t],",
+            "\t\t\"httpListeners\": [",
+            "\t\t\t{",
+            "\t\t\t\t\"name\": \"${16:appGatewayHttpListener}\",",
+            "\t\t\t\t\"properties\": {",
+            "\t\t\t\t\"FrontendIPConfiguration\": {",
+            "\t\t\t\t\t\"Id\": \"[resourceId('Microsoft.Network/applicationGateways/frontendIPConfigurations', '${1:applicationGatewayName}', '${7:appGatewayFrontendIP}')]\"",
+            "\t\t\t\t},",
+            "\t\t\t\t\"FrontendPort\": {",
+            "\t\t\t\t\t\"Id\": \"[resourceId('Microsoft.Network/applicationGateways/frontendPorts',  '${1:applicationGatewayName}', '${8:appGatewayFrontendPort}')]\"",
+            "\t\t\t\t},",
+            "\t\t\t\t\t\"Protocol\": \"${17|Http,Https|}\",",
+            "\t\t\t\t\t\"SslCertificate\": null",

```

Comment:
SslCertificate needs to be an object with a property of Id.

![image](https://user-images.githubusercontent.com/7844635/80731800-3a9dde80-8ac0-11ea-9e59-2324e43ee89f.png)

Change to something like:

![image](https://user-images.githubusercontent.com/7844635/80731907-602ae800-8ac0-11ea-9ca8-782b66be5a07.png)

Perhaps put a tab stop (18) between for the value of id and then update all following tab stops.

Reply:
Well then we need the protocol above to be https then, I suppose.
In the current scenario with http the template is correct as I deployed it. I will need to test with https. 
Cheers
Manu

Reply:
Sorry, yep makes sense. The other option would be to remove SslCertificate.

Reply:
hum, not sure.
As an ARM author, for the Frontend part it will likely be https (should be), but if use http and there is an id inside sslCertificate then the ARM validation will fail.
So out of the box like I proposed it works for both except that if you do choose https, then you have to specify further the cert.
Makes sense boss 😄 ?

Reply:
Just deploying again to confirm but App GW is not a fast provisioning resource  😝

Reply:
Confirmed that deployment fails if we have an empty id (null or "")

Nasty thing is that have a webApplicationFirewallConfiguration force a sku type WAF.
how do you prefer to deal with it?
2 snippets, one for regular App Gateways and one for WAFs ?

Cheers
Manu

Reply:
Thanks @ManuInNZ I am looking into this now. It may take a me a few days to play around / ramp up on the details of app gateway, but I will come back with something.